client/protocol.py

```diff
@@ -87,7 +87,7 @@ async def fast_path(self, cmd: Command) -> tuple[CommandResponse, SyncResponse |
         """
         for futures in asyncio.as_completed([self.fast_round(cmd), self.slow_round(cmd)]):
             first, second = await futures
-            if isinstance(first, CommandResponse) and isinstance(second, bool):
+            if isinstance(first, CommandResponse) and isinstance(second, bool) and second:
```

Comment:
It seems the `isinstance(second, bool)` is redundant because it is already checked in `isinstance(first, CommandResponse)`

Reply:
You're right. I've made the changes you suggested.